scripts/generator/graphql_generator/csharp/Cart.cs.erb

```diff
@@ -139,16 +154,35 @@ namespace <%= namespace %> {
                 CurrentCheckout = merger.Merge(CurrentCheckout, checkout);    
             }
 
+            if (userErrors.Count > 0) {
+                _UserErrors = userErrors;
+            } else {
+                _UserErrors = null;
+            }
+
             UpdateLineItemFromCheckout(CurrentCheckout);
+
+            return _UserErrors == null;
         }
 
         private void UpdateLineItemFromCheckout(Checkout checkout) {
-            // sometimes we may not query line items for instance when polling is being performed
-            try {
-                List<CheckoutLineItem> lineItems = (List<CheckoutLineItem>) checkout.lineItems();
+            if (checkout != null) {
+                // sometimes we may not query line items for instance when polling is being performed
+                try {
+                    List<CheckoutLineItem> lineItems = (List<CheckoutLineItem>) checkout.lineItems();
+
+                    LineItems.UpdateLineItemsFromCheckoutLineItems(lineItems);
+                } catch(NoQueryException exception) {}
+            }
+        }
+
+        private void HandleUserError(List<UserError> userErrors, Action<ShopifyError> callback) {
+            ShopifyError error = new ShopifyError(
+                ShopifyError.ErrorType.UserError,
+                "There were issues with some of the fields sent. Checkout `cart.UserErrors`"
```

Comment:
I think `UserErrors` as errors from a form. You might have multiple errors in the form at any given time. What error is most important? In the end the developer might want to handle all errors at the same time. I mentioned this issue in #105 that for `UserErrors` its tough to return any kind of meaningful data in the `description`.